.ci/gitlab/scripts/common/write_dotenv.py

```diff
@@ -0,0 +1,48 @@
+import os
+import sys
+
+CI_PROJECT_DIR = "."
+CI_AUX_DIR = os.path.join(".ci", "tmp", "spack")
+
+
+def configure_packages_repo(dotenv):
+    dotenv["SPACK_PACKAGES_ROOT"] = CI_PROJECT_DIR
+    dotenv["SPACK_ROOT"] = CI_AUX_DIR
+
+
+def configure_core_repo(dotenv):
+    dotenv["SPACK_PACKAGES_ROOT"] = CI_AUX_DIR
+    dotenv["SPACK_ROOT"] = CI_PROJECT_DIR
+
+
+def read_dotenv(file):
+    dotenv = {}
+    for line in file:
+        if "=" in line:
+            key, value = line.rsplit("=", 1)
+            dotenv[key] = value.strip()
+    return dotenv
+
+
+def write_dotenv(env, file):
+    for var in dotenv:
+        fd.write(f"{var}={dotenv[var]}\n")
```

Comment:
```suggestion
    for key, value in env.items():
        file.write(f"{key}={value}\n")
```
mismatch between `fd` and `file`, but also changed how we iterate for better readability.

Reply:
I am not sure how this was working...